### ISSUE TITLE
F11 (fullscreen) on Windows makes OBS blackscreen

### ISSUE THREAD

**Issue by qubodup** [olive-editor/olive#717]
Olive-f47a7dad-Windows-x86_64 gtx970 nvidia 419.35, 2 displays

Pressing F11 makes OBS lose the picture (display capture) and require restart.

If I restart OBS and click on other windows in the other display and then move over olive, it flickers black and OBS loses the picture again.

I'm guessing F11 does OpenGL magic (via Qt?). In games it is common that there is "fullscreen" (bad) and "borderless windowed" (good). If there's another fullscreen option in qt, it might be the better one.

Strangely, using borderlessgaming rather than olive's F11 has the exact same effect.

**Comment by itsmattkc** [comment]
Olive/Qt's fullscreen I'm quite certain is a borderless window. This seems like an issue you should bring up with OBS, especially since you say other software causes it to do the same thing.

**Comment by qubodup** [comment]
My bad, incomplete report: this (black flickering) happens without OBS too.

borderlessgaming is a tool to force fullscreen (remove window decoration, stretch to screen) in games and apps that don't have that feature. The fact that it makes olive behave the same way makes me guess it uses the same method that olive does or that something about olive makes it blackscreen for a frame or two when it's in fullscreen mode.

Here's a video of the behavior in olive, while obs and borderlessgaming are not turned on:
https://youtu.be/SI3yCO8F1OM

I have a somewhat similar issue with VLC which would cause blackscreen for more than one second each time I start/close it. that issue is solved by changing a Nvidia setting: https://www.youtube.com/watch?v=m0T0Oln6khk . No leads yet whether this is nvidia's fault.

I just realized that the same happens with kdenlive if I use its fullscreen mode. It does not happen with shotcut (which doesn't have a fullscreen mode) combined with borderlessgaming. Kdenlive seems to flicker each time it gets/loses focus and if it gets focus, the project monitor shows nothing.

So I tested this again in olive Olive-00e86a4e-Windows-x86_64 by adding a clip to the timeline and playing back and the project monitor works fine. However the entire window also now flickers each time focus is lost/regained.

Looks to be a qt issue.

Here's one workaround for v4: https://forum.qt.io/topic/87874/full-screen-makes-the-screen-flickers-on-windows-8/5

The google search term "qt fullscreen black flicker" (no quotes) seems to give relevant results.
